tex/gregoriotex-signs.tex

```diff
@@ -1644,19 +1644,21 @@
 % #3 is the char of the alteration hole
 % #4 is 1 in the case of a flat for a key change, 0 otherwise
 \def\gre@alteration#1#2#3#4{%
-  % to see why we reset \gre@lastoflinecount, see comments of \GreGlyph.
+  \setbox\gre@box@temp@width=\hbox{#2}%
   \ifnum#4=0\relax %
     \gre@newglyphcommon %
+    \ifgre@firstglyph%
+      \global\gre@dimen@bolextra = \wd\gre@box@temp@width%
+      \gre@skip@temp@four = \gre@skip@alterationspace\relax%
+      \global\advance\gre@dimen@bolextra by \dimexpr\gre@skip@temp@four\relax%
+      \gre@debugmsg{bolshift}{bolextra: \the\ger@diment@bolextra}%
+%			\kern -\gre@dimen@bolextra\relax%
+    \fi%
```

Comment:
Not really, in fact we need the alteration shift only if `bolshift > width(alteration)`, which will, I think, almost never be the case... In almost all cases, what we'll want is an alteration separated from the clef by `clefflatspace`. Do you see what I have in mind? If we do this, then there are two cases:

- the note is still to far right, well, at least we'll have done what we could, we cannot make it move left any further anyway, so it's ok
- the note is too far left, but I don't believe this case can happen...

do you see what I mean?